src/server/routes/obvius.js

```diff
@@ -0,0 +1,199 @@
+		'LOOPNAME', 'UPTIME', 'PERCENTBLOCKSINUSE', 'PERCENTINODESINUSE',
+		'UPLOADATTEMPT', 'ACQUISUITEVERSION', 'USRVERSION', 'ROOTVERSION',
+		'KERNELVERSION', 'FIRMWAREVERSION', 'BOOTCOUNT', 'BATTERYGOOD'];
+	// Build a log entry for this request
+	let s = `Handling request from ${ip}\n`;
+	for (const paramName of paramNames) {
+		if (req.param(paramName) !== false && req.param(paramName) !== undefined) {
+			s += `\tGot ${paramName}: ${req.param(paramName)}\n`;
+		} else {
+			s += `\tNo ${paramName} submitted\n`;
+		}
+	}
+	log.info(s);
+
+	success(req, res);
+}
+
+/**
+ * Handle an Obvius upload request.
+ * Unfortunately the Obvious API does not specify a HTTP verb.
+ */
+router.all('/', async (req, res) => {
+	// Log the IP of the requester
+	const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
+	log.info(`Received Obvious protocol request from ${ip}`);
+
+	// Attempt to verify the password
+	if (!req.param('password')) {
+		failure(req, res, 'password parameter is required.');
+		return;
+	} else if (req.param('password') !== config.obvius.password) {
+		failure(req, res, 'password was not correct.');
+		return;
+	}
+
+	const mode = req.param('mode', false);
+	if (mode === false) {
+		failure(req, res, 'Request must include mode parameter.');
+		return;
+	}
+
+	if (mode === obvius.mode.status) {
+		handleStatus(req, res);
+		return;
+	}
+
+	if (mode === obvius.mode.logfile_upload) {
+		if (!req.param('serialnumber', false)) {
+			failure(req, res, 'Logfile Upload Requires Serial Number');
+			return;
+		}
+		const conn = getConnection();
+		for (const fx of req.files) {
+			log.info(`Received ${fx.fieldname}: ${fx.originalname}`);
+			// Logfiles are always gzipped.
+			let data;
+			try {
+				data = zlib.gunzipSync(fx.buffer);
+			} catch (err) {
+				log.error(err);
+				failure(req, res, `Unable to gunzip incoming buffer: ${err}`);
+				return;
+			}
+			loadLogfileToReadings(req.param('serialnumber'), ip, data, conn);
+		}
+		success(req, res, 'Logfile Upload IS PROVISIONAL');
+		return;
+	}
+
+	if (mode === obvius.mode.config_file_download) {
+		failure(req, res, 'Config Download Not Implemented');
+		return;
+	}
+
+	if (mode === obvius.mode.config_file_manifest) {
+		const conn = getConnection();
+		success(req, res, await listConfigfiles(conn));
+		return;
+	}
+
+	if (mode === obvius.mode.config_file_upload) {
+		// Check required parameters
+		if (!req.param('serialnumber', false)) {
+			failure(req, res, 'Config Upload Requires Serial Number');
+			return;
+		}
+		if (!req.param('modbusdevice', false)) {
+			failure(req, res, 'Config Upload Requires Modbus Device ID');
+			return;
+		}
+		const conn = getConnection();
+		for (const fx of req.files) {
+			log.info(`Received ${fx.fieldname}: ${fx.originalname}`);
+
+			let data;
+			try {
+				data = zlib.gunzipSync(fx.buffer).toString('utf-8');
+			} catch (error) {
+				data = fx.buffer.toString('utf-8');
```

Comment:
This is treated differently when the gunzip fails for a log file. Should we be expecting a file that is not zipped? If not, this should be treated as in log upload.